### ISSUE TITLE
Video Classification issues

### ISSUE THREAD

**Issue by imneonizer** [PyTorchLightning/lightning-flash#721]
## 🐛 Bug

The example snippet in Readme seems to be broken
````python
import os
from torch.utils.data.sampler import RandomSampler
import flash
from flash.core.data.utils import download_data
from flash.video import VideoClassificationData, VideoClassifier

# 1. Download a video clip dataset. Find more datasets at https://pytorchvideo.readthedocs.io/en/latest/data.html
download_data("https://pl-flash-data.s3.amazonaws.com/kinetics.zip")

# 2. Load the Data
datamodule = VideoClassificationData.from_folders(
    train_folder=os.path.join(flash.PROJECT_ROOT, "data/kinetics/train"),
    val_folder=os.path.join(flash.PROJECT_ROOT, "data/kinetics/val"),
    predict_folder=os.path.join(flash.PROJECT_ROOT, "data/kinetics/predict"),
    batch_size=8,
    clip_sampler="uniform",
    clip_duration=1,
    video_sampler=RandomSampler,
    decode_audio=False,
    num_workers=8,
)

# 3. Build the model
model = VideoClassifier(backbone="x3d_xs", num_classes=datamodule.num_classes, pretrained=False)

# 4. Create the trainer
trainer = flash.Trainer(max_epochs=3)

# 5. Finetune the model
trainer.finetune(model, datamodule=datamodule)

# 6. Save it!
trainer.save_checkpoint("video_classification.pt")
````

Bydefault dataset is downloaded to current directory however the snipped points it to PROJECT_ROOT directory i.e.,
````
os.path.join(flash.PROJECT_ROOT, "data/kinetics/train")
````

After updating the correct path, another issue that I am facing currently is
````sh
/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'pl-flash-data.s3.amazonaws.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
GPU available: True, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py:1294: UserWarning: GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.
  rank_zero_warn(
Uncaught exception
Traceback (most recent call last):
  File "main.py", line 30, in <module>
    trainer.finetune(model, datamodule=datamodule)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/flash/core/trainer.py", line 157, in finetune
    return super().fit(model, train_dataloader, val_dataloaders, datamodule)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 552, in fit
    self._run(model)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 911, in _run
    self._pre_dispatch()
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 940, in _pre_dispatch
    self._log_hyperparams()
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 969, in _log_hyperparams
    self.logger.save()
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py", line 48, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/loggers/tensorboard.py", line 250, in save
    save_hparams_to_yaml(hparams_file, self.hparams)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/pytorch_lightning/core/saving.py", line 405, in save_hparams_to_yaml
    yaml.dump(v)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/__init__.py", line 290, in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/__init__.py", line 278, in dump_all
    dumper.represent(data)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 27, in represent
    node = self.represent_data(data)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 52, in represent_data
    node = self.yaml_multi_representers[data_type](self, data)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 342, in represent_object
    return self.represent_mapping(
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 118, in represent_mapping
    node_value = self.represent_data(item_value)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 52, in represent_data
    node = self.yaml_multi_representers[data_type](self, data)
  File "/home/nitin/github/video_classification/venv/lib/python3.8/site-packages/yaml/representer.py", line 330, in represent_object
    dictitems = dict(dictitems)
ValueError: dictionary update sequence element #0 has length 1; 2 is required
````

Environment
PyTorch Version (e.g., 1.0): 1.9.0
OS (e.g., Linux): Linux (Ubuntu 18.04)
How you installed PyTorch: pip
Python version: 3.8
CUDA/cuDNN version: 11.3
GPU models and configuration: GTX 1650
`lightning-flash==0.4.0`

**Comment by ethanwharris** [comment]
Hi @imneonizer we've just started encountering this issue too, currently trying to debug :smiley: In the meantime it's possible you could fix it by installing an older version of PyTorch-Lightning

**Comment by imneonizer** [comment]
`pytorch-lightning==1.3.0` is the latest one working for me

**Comment by ethanwharris** [comment]
Hey @imneonizer after some debugging I've narrowed it down to the recent release of torchmetrics, so you should be able to get away with just installing `torchmetrics<=0.5.0` to fix the error. I've opened PytorchLightning/metrics#492 to track it. On the flash side, I'll mark `torchmetrics==0.5.1` as incompatible

**Comment by tchaton** [comment]
Dear @imneonizer,

Any updates ?

Best,
T.C

**Comment by ethanwharris** [comment]
@tchaton actually this issue can be closed, we now only support `torchmetrics!=0.5.1` as that version has the error. @imneonizer please feel free to reopen this if you're issue isn't solved.

**Comment by imneonizer** [comment]
> Dear @imneonizer,
> 
> Any updates ?
> 
> Best,
> T.C

Yes changing the version did work out!
